toolkit.nu

```diff
@@ -31,36 +33,37 @@ export def fmt [
 # > it is important to make `clippy` happy :relieved:
 export def clippy [
     --verbose: bool # print extra information about the command's progress
-    --dataframe: bool # use the dataframe feature
+    --features: list<string> # the list of features to run *Clippy* on
 ] {
     if $verbose {
         print $"running ('toolkit clippy' | pretty-print-command)"
     }
 
-    try {
-        if $dataframe {
-            cargo clippy --workspace --features=dataframe,extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect -A clippy::result_large_err
-        } else {
-            cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect -A clippy::result_large_err
+    try {(
+        cargo clippy
+            --workspace
+            --features ($features | str join ",")
+        --
+            -D warnings
+            -D clippy::unwrap_used
+            -A clippy::needless_collect
+            -A clippy::result_large_err
+    )} catch {
+        error make --unspanned {
+            msg: $"\nplease fix the above ('clippy' | pretty-print-command) errors before continuing!"
         }
-    } catch {
-        error make -u { msg: $"\nplease fix the above ('clippy' | pretty-print-command) errors before continuing!" }
     }
 }
 
 # check that all the tests pass
 export def test [
     --fast: bool  # use the "nextext" `cargo` subcommand to speed up the tests (see [`cargo-nextest`](https://nexte.st/) and [`nextest-rs/nextest`](https://github.com/nextest-rs/nextest))
-    --dataframe: bool # use the dataframe feature
+    --features: list<string> # the list of features to run *Clippy* on
```

Comment:
```suggestion
    --features: list<string> # the list of features to run *test* on
```

Reply:
should be good in 37345155f872f792aa1ea570fb93ebb36ec9966a :relieved: